map/src/context/AppContext.js

```diff
@@ -327,7 +333,29 @@ export const AppContextProvider = (props) => {
     const [pinPoint, setPinPoint] = useState(pinInit);
     const [interPoints, setInterPoints] = useState([]);
     const [weatherPoint, setWeatherPoint] = useState(null);
-    const [favorites, setFavorites] = useState({file: null, groups: [], groupsUnique: [], readFirst: false});
+    const [favorites, setFavorites] = useState({
+        file: null,
+        groups: [],
+        groupsUnique: [],
+        readFirst: false,
+        visibleMarker: {
+            prev: null,
+            current: null
+        }
+    });
+    const [planRoute, setPlanRoute] = useState(false);
```

Comment:
what's difference between planRoute / newRoute / editRoute? sounds as same object

Reply:
planRoute - flag for mode when panelButton appear
newRoute - object
editor - object with edit functions